### PR TITLE
add hystrix circuit breaker

### PR DESCRIPTION
**Pull request by josler**
adding on top of https://github.com/go-kit/kit/pull/58

### CODE REVIEW

File: circuitbreaker/hystrix.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import (
+	"errors"
+
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/go-kit/kit/endpoint"
+	"golang.org/x/net/context"
+)
+
+var ErrCircuitBreakerTimeout = errors.New("circuit breaker timeout")
```
Comment (josler) [inline circuitbreaker/hystrix.go]: might want to rationalize this this with `ErrCircuitBreakerOpen` in `handy_breaker.go`

File: circuitbreaker/hystrix.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package circuitbreaker
+
+import (
+	"errors"
+
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/go-kit/kit/endpoint"
```
Comment (peterbourgon) [inline circuitbreaker/hystrix.go]: goimports style: move this import statement to a third group, separated by a blank line, please :)

File: circuitbreaker/hystrix.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+func wellKnownHystrixError(err error) error {
+	switch err {
+	case hystrix.ErrCircuitOpen:
+		return ErrCircuitBreakerOpen
+	case hystrix.ErrTimeout:
+		return ErrCircuitBreakerTimeout
+	default:
+		return err
+	}
+}
```
Comment (peterbourgon) [inline circuitbreaker/hystrix.go]: Any reason we shouldn't return these errors directly?

File: circuitbreaker/hystrix_test.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		if _, err := e(context.Background(), struct{}{}); err != circuitbreaker.ErrCircuitBreakerOpen {
+			t.Errorf("with request #%d, want %v, have %v", primeWith+letThru+i, circuitbreaker.ErrCircuitBreakerOpen, err)
+		}
+	}
+
+	// Confirm the rest didn't get through.
+	if want, have := primeWith+letThru, thru; want != have {
+		t.Errorf("want %d, have %d", want, have)
+	}
+}
```
Comment (peterbourgon) [inline circuitbreaker/hystrix_test.go]: We should factor a lot of this out to a general circuitbreaker test driver, but I'll do that afterwards.

File: circuitbreaker/hystrix.go (hunk 3 above)
Comment (josler) [inline circuitbreaker/hystrix.go]: Perhaps it'd be nice to return the same error type for all circuit breakers implemented? If that's not the desired behaviour we can just remove this :+1:

Comment (peterbourgon) [inline circuitbreaker/hystrix.go]: There may indeed be value in doing that, but I think I prefer the idea of defaulting to transparency, and just returning the errors directly. If it turns out users want consistent errors independent of concrete implementation, we can add it in by request, later.

**Comment by josler** [comment]
Updated - transparent errors, import cleanup

**Comment by josler** [comment]
Hm. Travis seems to not like running the tests concurrently.

**Comment by peterbourgon** [comment]
Thanks a bunch!

**Pull request merged**